components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandlerTest.java

```diff
@@ -0,0 +1,450 @@
+public class SAML2BearerGrantHandlerTest extends PowerMockTestCase {
+
+    private SAML2BearerGrantHandler saml2BearerGrantHandler;
+
+    private Assertion assertion;
+
+    private  ServiceProvider serviceProvider;
+
+    @Mock
+    private OauthTokenIssuer oauthIssuer;
+
+    @Mock
+    private OAuthComponentServiceHolder oAuthComponentServiceHolder;
+
+    @Mock
+    private RealmService realmService;
+
+    @Mock
+    private TenantManager tenantManager;
+
+    @Mock
+    private IdentityProviderManager identityProviderManager;
+
+    @Mock
+    private IdentityProvider identityProvider;
+
+    @Mock
+    private FederatedAuthenticatorConfig federatedAuthenticatorConfig;
+
+    @Mock
+    private SSOServiceProviderConfigManager ssoServiceProviderConfigManager;
+
+    @Mock
+    private OAuthTokenReqMessageContext tokReqMsgCtx;
+
+    @Mock
+    private OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO;
+
+    @Mock
+    private OAuthServerConfiguration oAuthServerConfiguration;
+
+    @Mock
+    private SAMLSignatureProfileValidator profileValidator;
+
+    @Mock
+    private X509Certificate x509Certificate;
+
+    @Mock
+    private SignatureValidator signatureValidator;
+
+    @Mock
+    private UserStoreManager userStoreManager;
+
+    @Mock
+    private UserRealm userRealm;
+
+
+    @Mock
+    private ApplicationManagementService applicationManagementService;
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getIdentityOauthTokenIssuer()).thenReturn(oauthIssuer);
+        saml2BearerGrantHandler = new SAML2BearerGrantHandler();
+        saml2BearerGrantHandler.init();
+        assertion = buildAssertion();
+    }
+
+    @Test
+    public void testValidateGrant() throws Exception {
+
+        oAuth2AccessTokenReqDTO = new OAuth2AccessTokenReqDTO();
+        saml2BearerGrantHandler = new SAML2BearerGrantHandler();
+        tokReqMsgCtx = new OAuthTokenReqMessageContext(oAuth2AccessTokenReqDTO);
+        tokReqMsgCtx.setTenantID(-1234);
+        String assertionString = SAMLSSOUtil.marshall(assertion);
+        assertionString = new String(Base64.encodeBase64(assertionString.getBytes(Charsets.UTF_8)));
+        oAuth2AccessTokenReqDTO.setAssertion(assertionString);
+        when(IdentityUtil.unmarshall(anyString())).thenReturn(assertion);
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        mockStatic(IdentityProviderManager.class);
+        mockStatic(IdentityProvider.class);
+        mockStatic(OAuthServerConfiguration.class);
+        mockStatic(IdentityApplicationManagementUtil.class);
+        IdentityProvider identityPro = new IdentityProvider();
+        identityPro.setIdentityProviderName("LOCAL");
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
```

Comment:
IMO the test method would look clean if we offload IDP creation logic to a separate private method.